support/spring-data-jpa-javax/src/main/kotlin/com/linecorp/kotlinjdsl/support/spring/data/jpa/javax/repository/KotlinJdslJpqlExecutorImpl.kt

```diff
@@ -174,8 +190,137 @@ open class KotlinJdslJpqlExecutorImpl(
         init: DSL.() -> JpqlQueryable<DeleteQuery<T>>,
     ): Int {
         val query: DeleteQuery<T> = jpql(dsl, init)
-        val jpaQuery = JpqlEntityManagerUtils.createQuery(entityManager, query, renderContext)
+        val jpaQuery = createJpaQuery(query)
 
         return jpaQuery.executeUpdate()
     }
+
+    private fun <T : Any> createJpaQuery(
+        query: JpqlQuery<*>,
+        returnType: KClass<T>,
+    ): TypedQuery<T> {
+        return JpqlEntityManagerUtils.createQuery(entityManager, query, returnType, renderContext).apply {
+            setMetadata(this, metadata)
+        }
+    }
+
+    private fun createJpaQuery(
+        query: JpqlQuery<*>,
+    ): Query {
+        return JpqlEntityManagerUtils.createQuery(entityManager, query, renderContext).apply {
+            setMetadata(this, metadata)
+        }
+    }
+
+    private fun <T : Any> createJpaEnhancedQuery(
+        query: JpqlQuery<*>,
+        returnType: KClass<T>,
+        sort: Sort,
+    ): EnhancedTypedQuery<T> {
+        val enhancedQuery = JpqlEntityManagerUtils.createEnhancedQuery(
+            entityManager,
+            query,
+            returnType,
+            sort,
+            renderContext,
+        )
+
+        return EnhancedTypedQuery(
+            enhancedQuery.sortedQuery.apply { setMetadata(this, metadata) },
+        ) {
+            // Lazy
+            enhancedQuery.countQuery.apply { setMetadataForCount(this, metadata) }
+        }
+    }
+
+    private fun setMetadata(query: Query, metadata: CrudMethodMetadata?) {
+        if (metadata == null) return
+
+        setLockMode(query, metadata.lockModeType)
+        setQueryHints(query, metadata.queryHints)
+    }
+
+    private fun setMetadataForCount(query: Query, metadata: CrudMethodMetadata?) {
+        if (metadata == null) return
+
+        setQueryHints(query, metadata.queryHintsForCount)
+    }
+
+    private fun setLockMode(query: Query, lockMode: LockModeType?) {
+        if (lockMode != null) {
```

Comment:
```suggestion
        lockMode?.run {
```
how about this style?

Reply:
I think `if` is better in this case.

Reply:
ok. It's a style so it doesn't really matter
However, for null checking, how about !== null (reference comparison) rather than != null?

Reply:
How do they differ? What's the difference between the two? I checked with Java compilation and there is no difference between `lockMode?.run`, `if (lockMode != null)` and `if (lockMode !== null)` for null checking.

Reply:
== is an equals comparison, === is a reference comparison.
However, as far as I know, this is according to the Kotlin language specification and may be the same when converted to Java.
In the case of variable?.run , the run method is an inline method, so we expect the code to be converted to Java code as a result, similar to the if code.

As a result, === conveys a clear intent about which values require reference comparison.

Reply:
You're not talking about functionality, you're talking about how to express the code. I think it's not a bad idea to use reference equality to indicate whether the reference of this object is null or not. But if you have to enforce it for the whole project, I don't think so. 

Kotlin provides both structural equality and reference equality as ways of checking nullability. You can see this by looking at the code that Kotlin itself uses. They use structural equality in some places and reference equality in others to check the nullability.

[==null](https://github.com/search?q=org%3AKotlin+%3D%3D+null+language%3AKotlin&type=code&l=Kotlin) case
[===null](https://github.com/search?q=org%3AKotlin+%3D%3D%3D+null+language%3AKotlin&type=code) case

So I don't want to force people to use === when checking the nullability.

Reply:
I appreciate your respect.

Regarding enums, I would rather use == than ===. Enums are singletons, but they are also objects. We use structural equality a lot as a way to compare two objects. That's why I think it's better to use == when comparing enums.

For example, suppose you have a method that looks like this:
```kotlin
fun compare(color1: Color, color2: Color): Boolean {
  TODO()
}
```

How would you compare them? When you use ===, it means that you know that the Color class is implemented as an Enum class.

What would happen if you changed the implementation of the Color class to a Sealed class instead of an Enum class? If you used ===, you would have to change the implementation.

This is because Enum is a way to implement an enumeration, and Kotlin provides Sealed classes as a way to implement an enumeration. That's why I prefer to use === instead of ==.

Reply:
It's like a difference in perspective. If you change to a sealed class, there is a possibility that the logic will change a lot, and you will definitely have to recheck the automated tests you created, but I don't think there is a need to approach the issue of whether or not to fix the code. However, in the case of null checking inside Kotlin, there is documentation that === or == will be converted automatically, so there is no harm in following your convention unless there are other cases where you need to explicitly check the reference.

Reply:
Wow... that's amazing. I'm learning.